src/Text/LLVM/PP.hs

```diff
@@ -1235,11 +1235,34 @@ ppDISubprogram' pp sp = "!DISubprogram"
 ppDISubprogram :: Fmt DISubprogram
 ppDISubprogram = ppDISubprogram' ppLabel
 
+ppDISubrange' :: Fmt i -> Fmt (DISubrange' i)
+ppDISubrange' pp sr =
+  "!DISubrange"
+  <> parens (mcommas
+             -- LLVM < 7: count and lowerBound as signed int 64
+             -- LLVM < 11: count as ValMd, lowerBound as signed in 64
+             -- LLVM >= 11: ValMd of count, lowerBound, upperBound, and stride
+             -- Valid through LLVM 17.
+             -- See AST.hs description for more details.
+             [
+               let ppV = if llvmVer >= 7
+                         then ppValMd' pp
+                         else ppInt64ValMd'
+               in (("count:" <+>) . ppV) <$> disrCount sr
+             , let ppV = if llvmVer >= 11
+                         then ppValMd' pp
+                         else ppInt64ValMd'
+               in (("lowerBound:" <+>) . ppV) <$> disrLowerBound sr
+             , if llvmVer >= 11
+               then (("upperBound:" <+>) . ppValMd' pp) <$> disrUpperBound sr
+               else Nothing
+             , if llvmVer >= 11
+               then (("stride:" <+>) . ppValMd' pp) <$> disrStride sr
+               else Nothing
```

Comment:
We could use [`when'`](https://github.com/elliottt/llvm-pretty/blob/3a9599072cb9fa59e9ba9321a9e46030ce975f43/src/Text/LLVM/PP.hs#L97-L100) here instead if these `if ... then ... else Nothing` constructs.

Reply:
Done.